src/dict.c

```diff
@@ -820,9 +820,9 @@ void dictSetKey(dict *d, dictEntry* de, void *key) {
         de->key = key;
 }
 
-void dictSetVal(dict *d, dictEntry *de, void *val) {
```

Comment:
Trust in the compiler! or "respect and suspect" and actually examine the assembly. But from my experience the compiler is way smarter than mere humans and will inline this and a lot more where it make sense.

Reply:
If cross-file optimization is turned on, then I would expect a good compiler to automatically inline this.  We have a lot of `in-line` directives in our code, I cannot tell you why.

I am more interested in changing `assert` to `debugServerAssertWithInfo` so that it is a no-op in production.

I have a zero-trust policy.  I am willing to work with a "trust and verify" policy on the grounds that it really means "don't tell someone you don't trust them until after you have proof".  I think that is equivalent to your "respect and suspect".